src/driver/xen.rs

```diff
@@ -71,18 +77,40 @@ impl Xen {
         let (_ring_page, back_ring, remote_port) = xc
             .monitor_enable(cand_domid)
             .expect("Failed to map event ring page");
-        let xev = XenEventChannel::new(cand_domid, remote_port).unwrap();
 
+        // enable singlestep monitoring
+        // it will only intercept events when explicitely requested using
+        // xc_domain_debug_control()
+        // TODO: call get_vcpu_count()
```

Comment:
That comment is obsolete afais, right?

Reply:
Actually no.
I copy pasted the get_vcpu_count() implementation here, because if i created  the Xen struct below and tried to call get_vcpu_count() afterwards, it had an issue with `xc` being moved above when constructing the Xen struct.

As I didn't have any solution, I used this.

I suggest we should make a function get_vcpu_count(xc: &Xc) and call it from the trait and the Xen driver init function

Reply:
I've just refactored it.

Reply:
Hm, I don't see the problem ![this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=661acd93eb80b05457b11da953d02b41) works fine for me. On a side note, this function has become too large imho, we should split it.

Reply:
The problem was that I copy pasted a chunk of code, so there was duplication.
I we update the definition of `get_vcpu_count` one day, the other one becomes obselete, etc.

>  This works fine for me. 

Github won't diplay your link: Non-Image content-type returned

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=661acd93eb80b05457b11da953d02b41

Reply:
oh.
I don't know I was seeing having a mutable Xen struct as a bad idea.
it's not, since it's restricted on the init function anyway.

I refactored and implemented your changes, thank you